frontend/src/container/TimeSeriesView/TimeSeriesView.tsx

```diff
@@ -13,31 +15,45 @@ function TimeSeriesView({
 	isError,
 	yAxisUnit,
 }: TimeSeriesViewProps): JSX.Element {
-	const chartData = useMemo(
-		() =>
-			getChartData({
-				queryData: [
-					{
-						queryData: data?.payload?.data?.result || [],
-					},
-				],
-			}),
-		[data?.payload?.data?.result],
-	);
+	const graphRef = useRef<HTMLDivElement>(null);
+
+	const chartData = useMemo(() => getUPlotChartData(data?.payload), [
+		data?.payload,
+	]);
+
+	const isDarkMode = useIsDarkMode();
+
+	const width = graphRef.current?.clientWidth
+		? graphRef.current.clientWidth
+		: 700;
+
+	const height = graphRef.current?.clientWidth
+		? graphRef.current.clientHeight
+		: 300;
+
+	const chartOptions = getUPlotChartOptions({
+		yAxisUnit: yAxisUnit || '',
+		apiResponse: data?.payload,
+		dimensions: {
+			width,
+			height,
+		},
+		isDarkMode,
+	});
```

Comment:
This is the same code we are using in `FullView` [here](https://github.com/SigNoz/signoz/blob/2d11aa7fc6f97b54490c0125c25a63d58f13e0c1/frontend/src/container/GridCardLayout/GridCard/FullView/index.tsx#L97C5-L97C5) as well, Can we create a Custom hook for which can take ref as input and return the chart options?

Reply:
It's just return height and width, the fallback height and width can be set up case by case.